src/test/integration/integration-test.py

```diff
@@ -0,0 +1,92 @@
+            cls.session = utils.setup_session(config)
+            cls.test_cases = config['test_cases']
+
+        with open(openapi_path) as openapi_file:
+            cls.openapi = yaml.load(openapi_file)
+
+    @classmethod
+    def cleanup(cls):
+        cls.session.close()
+
+    # Test case: GET /spendingauthority with valid authority onids
+    def test_get_authority_onids(self, endpoint='/spendingauthority'):
+        for onid in self.test_cases['valid_authority_onids']:
+            params = {'onid': onid}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            spending_schema = utils.get_resource_schema(
+                self, 'SpendingAuthorityResource'
+            )
+            utils.check_schema(self, response, spending_schema)
+            response_data = response.json()['data']
+            actual_onid = response_data['id']
+            self.assertEqual(actual_onid.lower(), onid.lower())
+
+    # Test case: GET /spendingauthority with invalid authority onids
+    def test_get_non_authority_onids(self, endpoint='/spendingauthority'):
+        for onid in self.test_cases['invalid_authority_onids']:
+            params = {'onid': onid}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            spending_schema = []
+            self.assertEqual(response.json()['data'], spending_schema)
+
+    # Test case: GET /spendingauthority with bad request
+    def test_error_response(self, endpoint='/spendingauthority'):
+        for onid in self.test_cases['bad_request']:
+            params = {'onid': onid}
+            response = utils.make_request(self, endpoint, 400, params=params)
+            error_schema = utils.get_resource_schema(self, 'Error')
+            utils.check_schema(self, response, error_schema)
+
+    # Test case: GET /spendingauthority spending limits
+    def test_spending_limits(self, endpoint='/spendingauthority'):
+        # Checking that the lists of spending limits and indexes are unique
+        # Checking valid onids with spending authority only
+        for onid in self.test_cases['valid_authority_onids']:
+            params = {'onid': onid}
+            response = utils.make_request(self, endpoint, 200, params=params)
+            response_data = response.json()['data']
+            attributes = response_data['attributes']
+            limits = attributes['limits']
+            spendingLimit = []
+            index_array = []
+            # Checking that each spending limit has at least one index
+            for limit in limits:
+                for index in limit['indexes']:
+                    index_array.append(index)
+                spendingLimit.append(limit['spendingLimit'])
+                self.assertGreater(len(limit['indexes']), 0)
+            # Creating a dict indexes_list from indexes,
+            # and compare it with the original array of indexes
+            indexes_list = list(dict.fromkeys(index_array))
```

Comment:
Same naming convention issue here. Also, I could be wrong but I'm not exactly sure what did you want to do for `list(dict.fromkeys(index_array))`? Why do you want to transfer a `list` to a `dict` with all `None` values then change that back into a new `list` which is exactly the same as original list?

Reply:
To make sure that the list is unique, im putting the list in a dict, since we cant have duplicate keys in a dictionary, then compare the dict with the original list, I tried to add a duplicate key in the original list then transferring it to the dict to make sure that the duplicate key will be removed and it worked.

Reply:
So my question will be, why it's possible to have duplicated keys in the list? If there are any duplicated keys, we definitely don't wanna ignore them silently.

Reply:
They are not ignored, when comparing the list with the dict in `self.assertEqual(index_array, indexes_list)` we will catch the difference.

Reply:
Oh ok, I got it now. In this case, using a [set](https://en.wikipedia.org/wiki/Set_(abstract_data_type)) if more proper then using a dictionary in Python. To make sure there are no duplicated keys in the list, you can do something like:

```python
len(your_list) != len(set(your_list))
```